File: examples/nodejs/basic_usage.js
#!/usr/bin/env node
/**
 * MemoryStack - Node.js Basic Usage Example
 *
 * Shows how to use MemoryStack with the OpenAI Node.js SDK.
 * Just change the baseURL and your AI gets perfect memory!
 *
 * Installation:
 *   npm install openai
 *
 * Usage:
 *   node basic_usage.js
 */

const OpenAI = require('openai');

// Initialize client pointing to MemoryStack
const client = new OpenAI({
  apiKey: 'sk-your-openai-api-key',  // Your actual OpenAI API key
  baseURL: 'http://localhost:8000/v1',  // Point to MemoryStack
  defaultHeaders: {
    'X-User-Id': 'david'  // Optional: Namespace memories per user
  }
});

async function main() {
  console.log('=== First Conversation ===');

  // First conversation - store some information
  const response1 = await client.chat.completions.create({
    model: 'gpt-4o-mini',
    messages: [
      { role: 'user', content: 'My name is David and I love JavaScript. I build web applications.' }
    ]
  });

  console.log(`AI: ${response1.choices[0].message.content}\n`);

  // Second conversation - test memory
  console.log('=== Second Conversation (Different Chat) ===');

  const response2 = await client.chat.completions.create({
    model: 'gpt-4o-mini',
    messages: [
      { role: 'user', content: 'What\'s my name and what do I love?' }
    ]
  });

  console.log(`AI: ${response2.choices[0].message.content}\n`);
  // Expected: "Your name is David and you love JavaScript"

  // Third conversation - deeper question
  console.log('=== Third Conversation (Needs Long-term Memory) ===');

  const response3 = await client.chat.completions.create({
    model: 'gpt-4o-mini',
    messages: [
      { role: 'user', content: 'What kind of applications do I build?' }
    ]
  });

  console.log(`AI: ${response3.choices[0].message.content}\n`);

  console.log('✅ Memory working! Your AI remembers across different chats.');
}

main().catch(console.error);

File: examples/python/basic_usage.py
#!/usr/bin/env python3
"""
MemoryStack - Basic Usage Example

This example shows the simplest way to use MemoryStack with the OpenAI SDK.
Just change the base_url and your AI gets perfect memory!
"""

from openai import OpenAI

# Initialize client pointing to MemoryStack
client = OpenAI(
    api_key="sk-your-openai-api-key",  # Your actual OpenAI API key
    base_url="http://localhost:8000/v1",  # Point to MemoryStack instead
    default_headers={
        "X-User-Id": "alice"  # Optional: Namespace memories per user
    }
)

# First conversation - store some information
print("=== First Conversation ===")
response = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[
        {"role": "user", "content": "My name is Alice and I'm learning Python. I'm building a chatbot."}
    ]
)
print(f"AI: {response.choices[0].message.content}\n")

# Check memory metadata in response headers
# MemoryStack adds diagnostic headers like:
# X-Memory-Tiers-Used, X-Memory-Tier1-Turns, X-Memory-Processing-Time-Ms
# These are visible in HTTP responses but not in SDK responses

# Second conversation - test memory
print("=== Second Conversation (Different Chat) ===")
response = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[
        {"role": "user", "content": "What's my name and what am I working on?"}
    ]
)
print(f"AI: {response.choices[0].message.content}\n")
# Expected: "Your name is Alice and you're building a chatbot using Python"

# Third conversation - deeper question
print("=== Third Conversation (Needs Long-term Memory) ===")
response = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[
        {"role": "user", "content": "Tell me everything you know about me"}
    ]
)
print(f"AI: {response.choices[0].message.content}\n")

print("✅ Memory working! Your AI remembers across different chats.")
